opennebula/resource_opennebula_virtual_machine.go

```diff
@@ -266,6 +286,29 @@ func resourceOpennebulaVirtualMachineCreate(d *schema.ResourceData, meta interfa
 		}
 	}
 
+	// If the user requested a desired state except running at creation, handle this request after starting the VM
+	desiredState := d.Get("desired_state").(string)
```

Comment:
it is optional so you must test it before and you must do something only if is is provided